### PR TITLE
Eve cluster part2: Add EveWeather attributes

### PR DESCRIPTION
**Pull request by lboue**
Add Class attribute for Eve Cluster (EveWeather device):
- Altitude
- Pressure

HA core PR: [Add EveWeatherPressure attribute #123565](https://github.com/home-assistant/core/pull/123565)

### CODE REVIEW

**Review by agners — APPROVED** [review]

File: matter_server/common/custom_clusters.py [hunk 1]
```diff
@@ -243,6 +245,52 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
 
             value: float32 = 0
 
+        @dataclass
+        class Altitude(ClusterAttributeDescriptor, CustomClusterAttributeMixin):
+            """Altitude Attribute within the Eve Cluster."""
+
+            should_poll = True
```
Comment (agners) [inline matter_server/common/custom_clusters.py:252]: Are you sure this need to be polled as well? :thinking:

Comment (lboue) [reply matter_server/common/custom_clusters.py:252]: No, that's not necessary.

Comment (agners) [reply matter_server/common/custom_clusters.py]: I didn't meant to remove altitude, just use the subscription mechanism (if subscriptions work on these attributes, that you'd have to test on an actual device).

Comment (lboue) [reply matter_server/common/custom_clusters.py]: How can I manually test the subscription mechanism with my device?

Comment (agners) [reply matter_server/common/custom_clusters.py]: You can just set the log level to verbose, then you'll see what is happening. The `Attribute updated` string comes from the subscription callback: 

https://github.com/home-assistant-libs/python-matter-server/blob/c09db74df9f6ce43a37cff88a148069a0c15fd50/matter_server/server/device_controller.py#L1056-L1067

I actually bought such a device too, and I can see that the altitude gets updated via the subscription:
```
2024-08-16 14:39:29.781 (MainThread) VERBOSE [matter_server.server.device_controller] <Node:264> Attribute updated: 1/319486977/319422484 - old value: 959.7000122070312 - new value: 959.0999755859375
```

So in other words, we can set polled to false here, and it will still update :tada:

File: matter_server/common/custom_clusters.py [hunk 2]
```diff
@@ -243,6 +245,52 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
...
+
+            @ChipUtility.classproperty
+            def attribute_id(cls) -> int:
+                """Return attribute id."""
+                return 0x130A0013
+
+            @ChipUtility.classproperty
+            def attribute_type(cls) -> ClusterObjectFieldDescriptor:
+                """Return attribute type."""
+                return ClusterObjectFieldDescriptor(Type=int)
```
Comment (agners) [inline matter_server/common/custom_clusters.py:267]: Actually, it seems that this attribute is a float32 (from my logs):

```
TypeError: Value 0.0 of type <class 'float'> is invalid for altitude, expected value of type <class 'int'>
```

However, I see that the value currently stays at `0`. I don't think it is useful currently, so I think it is better to  just drop it until it actually provides useful information. Sorry for the back and fourth.

File: matter_server/common/custom_clusters.py [hunk 3]
```diff
@@ -243,6 +245,52 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
...
+
+            value: int = 0
+
+        @dataclass
+        class Pressure(ClusterAttributeDescriptor, CustomClusterAttributeMixin):
+            """Pressure Attribute within the Eve Cluster."""
+
+            should_poll = True
```
Comment (agners) [inline matter_server/common/custom_clusters.py:275]: As discussed in the other review, this does not need to be polled. The default is `False`, you can just drop it:

```suggestion
```

**Review by agners — CHANGES_REQUESTED** [review]

File: matter_server/common/custom_clusters.py (hunk 2 above)
Comment (lboue) [reply matter_server/common/custom_clusters.py:267]: In fact, it was set to zero because I hadn't configured it in the Eve application in the first place. Since then, the value is different from `0`

Comment (agners) [reply matter_server/common/custom_clusters.py:267]: Uh I see, this is more a configuration value :thinking: Yeah that make sense, it is probably used to calibrate/improve read out of the pressure sensor.

Ideally we would add it as a configuration then to HA.

Comment (lboue) [reply matter_server/common/custom_clusters.py:267]: image

File: matter_server/common/custom_clusters.py [hunk 4]
```diff
@@ -99,6 +99,8 @@ def descriptor(cls) -> ClusterObjectDescriptor:
     wattAccumulatedControlPoint: float32 | None = None
     voltage: float32 | None = None
     current: float32 | None = None
+    altitude: int | None = None
```
Comment (agners) [inline matter_server/common/custom_clusters.py:102]: Altitude is in fact float32 too, so:

```suggestion
    altitude: float32 | None = None
```

File: matter_server/common/custom_clusters.py [hunk 5]
```diff
@@ -243,6 +245,50 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
...
+
+            @ChipUtility.classproperty
+            def attribute_id(cls) -> int:
+                """Return attribute id."""
+                return 0x130A0013
+
+            @ChipUtility.classproperty
+            def attribute_type(cls) -> ClusterObjectFieldDescriptor:
+                """Return attribute type."""
+                return ClusterObjectFieldDescriptor(Type=int)
```
Comment (agners) [inline matter_server/common/custom_clusters.py:267]:
```suggestion
                return ClusterObjectFieldDescriptor(Type=float32)
```

File: matter_server/common/custom_clusters.py [hunk 6]
```diff
@@ -243,6 +245,50 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
...
+
+            value: int = 0
```
Comment (agners) [inline matter_server/common/custom_clusters.py:269]:
```suggestion
            value: float32 = 0
```

**Review by agners — CHANGES_REQUESTED** [review]
And finally, you also need to adjust it here: https://github.com/home-assistant-libs/python-matter-server/blob/1c3814722070295bfb349bb50b4fc81a8b7b3e1a/matter_server/common/custom_clusters.py#L88

**Comment by lboue** [comment]
> And finally, you also need to adjust it here:
> 
> https://github.com/home-assistant-libs/python-matter-server/blob/1c3814722070295bfb349bb50b4fc81a8b7b3e1a/matter_server/common/custom_clusters.py#L88

Fixed.

File: matter_server/common/custom_clusters.py [hunk 7]
```diff
@@ -243,6 +245,50 @@ def attribute_type(cls) -> ClusterObjectFieldDescriptor:
 
             value: float32 = 0
 
+        @dataclass
+        class Altitude(ClusterAttributeDescriptor, CustomClusterAttributeMixin):
+            """Altitude Attribute within the Eve Cluster."""
+
+            should_poll = False
+
```
Comment (agners) [inline matter_server/common/custom_clusters.py:253]: We can drop this too.

```suggestion
```

**Review by agners — APPROVED** [review]
Other than this, LGTM! Thanks!

**Comment by lboue** [comment]
Working fine with [6.5.0 BETA 1](https://github.com/home-assistant-libs/python-matter-server/releases/tag/6.5.0b1) version:

image
